src/shims/unix/sync.rs

```diff
@@ -118,13 +118,29 @@ fn mutex_get_id<'tcx>(
     ecx: &mut MiriInterpCx<'tcx>,
     mutex_op: &OpTy<'tcx>,
 ) -> InterpResult<'tcx, MutexId> {
+    let address = ecx.deref_pointer_as(mutex_op, ecx.libc_ty_layout("pthread_mutex_t"))?.ptr();
```

Comment:
```suggestion
    // FIXME: might be worth changing mutex_get_or_create_id to take the mplace
    // rather than the `OpTy`.
    let address = ecx.deref_pointer_as(mutex_op, ecx.libc_ty_layout("pthread_mutex_t"))?.ptr();
```